lib/core/src/components/button/Button.vue

```diff
@@ -0,0 +1,18 @@
+<template>
+  <button>{{ label }}</button>
+</template>
+
+<script lang="ts">
+import { defineComponent, ref } from "vue";
+export default defineComponent({
```

Comment:
Hm I'm not sure if we can use named export with Vue. Generally I'd prefer named exports to avoid implicit naming. (i.e. if the users want to use Button as some other names, they must explicitly use named imports)

Reply:
Maybe you want something like

```
const Button = defineComponent({
...

export { Button } 
```

right?

Reply:
Agree on this one, i prefer named export which is easier for vscode auto import

Reply:
yub ^

Reply:
* yup

Reply:
Other option:
```
@Options({
	...
})
export default class Component extends Vue {}
```

I don't know which one is better.

Reply:
Please refer here @thien-do [https://github.com/moaijs/moai-vue/blob/143d98db3a7d189e477fc36048592fc62f57feb2/lib/core/src/components/button/index.ts#L16](https://github.com/moaijs/moai-vue/blob/143d98db3a7d189e477fc36048592fc62f57feb2/lib/core/src/components/button/index.ts#L16)

Already required to use named import, to import a component, we'll need to import like this [https://github.com/moaijs/moai-vue/blob/143d98db3a/lib/docs/src/stories/components/Button.stories.js#L1](https://github.com/moaijs/moai-vue/blob/143d98db3a/lib/docs/src/stories/components/Button.stories.js#L1)

Reply:
But most of the time, users won't do that.

They will just:
```javascript
import Moai from '@moai-vue/core';

app.use(Moai);
```

Then use in template:

```vue
<template>
   <m-button></m-button>
</template>
```

Reply:
yeah that sounds good!

Reply:
Oh #TIL: https://github.com/vuejs/vue-loader/issues/1234

```
Features like hot module replacement, CSS injection, scoped CSS, SSR critical CSS collection all depend on being able to precisely locate the exported component - it will not work if the component can be exported as an arbitrarily named component. In addition, there are tools (ESLint plugin, for example) that are based on the assumption of the component being the default export.

TL;DR: the component must be the default export and this is not going to change.
```